### PR TITLE
how-to: move k8s guide into a dedicate how-to md

### PR DESCRIPTION
**Pull request by marcov**
k8s how-to: rewrite intro and introduce cri-o RuntimeClass
    Update the k8s Kata guide introduction, and specify RuntimeClass option
    for CRI-O 1.12+.

    how-to: improve landing README.md
    Improve landing README.md by categorizing howto guides.

    how-to: move k8s guide into a dedicate how-to md
    Create a dedicate how-to guide for running Kata with k8s, and link to it
    from the original guide location inside the Developer Guide.

 Fixes: #333

 Signed-off-by: Marco Vedovati <mvedovati@suse.com>

### CODE REVIEW

File: how-to/README.md [hunk 1]
```diff
@@ -1,11 +1,22 @@
-# Howto guides
+# HowTo Guides
```
Comment (jodh-intel) [inline how-to/README.md]: The spell checker won't like this I'm afraid. If you think we need the camel-case version, please raise a PR to add it to:

- https://github.com/kata-containers/tests/blob/master/cmd/check-spelling/data/main.txt

Comment (marcov) [reply how-to/README.md]: Let's stick with `Howto`  for now :laughing:

**Comment by marcov** [comment]
@jodh-intel I dont understand why the check-markdown is complainig about:
```
FATA[0000] file="how-to/run-kata-with-k8s.md": failed to find heading for link "kubernetes-runtimeclass-cri-o-v112" ({Doc:how-to/run-kata-with-k8s.md Address:kubernetes-runtimeclass-cri-o-v112 ResolvedPath: Description:Kubernetes RuntimeClass (CRI-O v1.12 ) Type:internal-link})  commit=5d201affb91ac8446d2a7a86b86deab517af8d85 name=kata-check-markdown pid=6601 source=check-markdown version=0.0.1
```

**Comment by marcov** [comment]
> @jodh-intel I dont understand why the check-markdown is complainig about:
> 
> ```
> FATA[0000] file="how-to/run-kata-with-k8s.md": failed to find heading for link "kubernetes-runtimeclass-cri-o-v112" ({Doc:how-to/run-kata-with-k8s.md Address:kubernetes-runtimeclass-cri-o-v112 ResolvedPath: Description:Kubernetes RuntimeClass (CRI-O v1.12 ) Type:internal-link})  commit=5d201affb91ac8446d2a7a86b86deab517af8d85 name=kata-check-markdown pid=6601 source=check-markdown version=0.0.1
> ```

It's failing because the markdown checker does not like this:
````
> ```
> codeblock
> ```

# New section
````

Replace it with:
`````
> ````
> codeblock
> ````

# New section
`````

**Comment by jodh-intel** [comment]
@marcov - ugh. That might be another bug with the blackfriday markdown parser we use (our tool already has to work around some issues with that package).

**Comment by grahamwhaley** [comment]
the ` > ``` ` could be a render issue. Kudos @marcov for knowing about ` ```` ` ;-) - we have only had to use that once before I think to code block ` ``` `'s :-)

File: how-to/README.md [hunk 2]
```diff
@@ -1,12 +1,22 @@
-# Howto guides
+# Howto Guides
 
+* [Howto Guides](#howto-guides)
```
Comment (grahamwhaley) [inline how-to/README.md]: Nice split out - thanks! James and I have always wondered why we have both howto and usecases - and have wondered if they should all go in one place ;-)

File: how-to/run-kata-with-k8s.md [hunk 3]
```diff
@@ -1,64 +1,86 @@
...
      * [Configure for CRI-O](#configure-for-cri-o)
      * [Configure for containerd](#configure-for-containerd)
   * [Run a Kubernetes pod with Kata Containers](#run-a-kubernetes-pod-with-kata-containers)
 
-## Pre-requisites
-This guide requires Kata Containers available on your system, and it can be installed
-following [this guide](https://github.com/kata-containers/documentation/blob/master/install/README.md).
-
+## Prerequisites
+This guide requires Kata Containers available on your system, install-able by following [this guide](https://github.com/kata-containers/documentation/blob/master/install/README.md).
```
Comment (grahamwhaley) [inline how-to/run-kata-with-k8s.md]: I wonder if we should point this at [kata-deploy](https://github.com/kata-containers/packaging/tree/master/kata-deploy), as:
- this is afaik the recommended way to install kata on k8s clusters
- you don't actually need kata installed on the master node (if it is not schedulable), or the host (where you run `kubectl` for instance), just the schedulable nodes that will run pods...

File: how-to/run-kata-with-k8s.md [hunk 4]
```diff
@@ -1,64 +1,86 @@
...
+[crio.runtime.runtimes.kata-qemu]
+  runtime_path = "/usr/bin/kata-runtime"
+  runtime_type = "oci"
+
+[crio.runtime.runtimes.kata-fc]
+  runtime_path = "/usr/bin/kata-runtime"
+  runtime_type = "oci"
+```
+
+#### Untrusted annotation (until CRI-O v1.12)
```
Comment (grahamwhaley) [inline how-to/run-kata-with-k8s.md]: I suspect we can and should drop the whole 'untrusted annotation' stuff from all of our docs (and, code if there is any). It has been replaced by the `RuntimeClass`...

Comment (marcov) [reply how-to/run-kata-with-k8s.md]: Thank you @grahamwhaley . 
There's also the  `## Run a Kubernetes pod with Kata Containers`  section 
that still needs to be updated with RuntimeClass, but I prefer to postpone that in a next PR and re-use the [Testing Kata Containers](https://github.com/kata-containers/documentation/blob/master/install/minikube-installation-guide.md#testing-kata-containers) in the minikube install guide.

**Comment by jodh-intel** [comment]
We need a review over here folks.

**Pull request merged**